frame/balances/src/lib.rs

```diff
@@ -991,6 +1006,19 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				);
 			}
 		}
+
+		let deposit_lock_event = |prev: T::Balance, after: T::Balance, reason: Reasons| {
+			if prev < after {
+				let amount = after.saturating_sub(prev);
```

Comment:
>Is it okay to use non-saturating math here or is using saturating math only enforced meanwhile?

Would be okay here. But since you already have it - its doing no harm.